nexus/utils/socialMedia.ts

```diff
@@ -0,0 +1,109 @@
+import { SocialMedia, SocialMediaPlatform } from '@prisma/client'
+import { toUpper } from 'lodash'
+import { InvalidInput } from '../errors'
+
+const socialMediaPatterns = [
+  {
+    platform: SocialMediaPlatform.FACEBOOK,
+    pattern: /^(?:(?:http|https):\/\/)?(?:www\.)?(?:facebook\.com|fb\.com)\/([A-Za-z0-9-_\.]+)\/?$/im,
+  },
+  {
+    platform: SocialMediaPlatform.INSTAGRAM,
+    pattern: /^(?:(?:http|https):\/\/)?(?:www\.)?(?:instagram\.com|instagr\.am)\/([A-Za-z0-9-_\.]+)\/?$/im,
+  },
+  {
+    platform: SocialMediaPlatform.LINKEDIN,
+    pattern: /^(?:(?:http|https):\/\/)?(?:www\.)?linkedin.com\/((in\/[^\/]+\/?)|(pub\/[^\/]+\/((\\w|\\d)+\/?){3}))$/im,
+  },
+  {
+    platform: SocialMediaPlatform.YOUTUBE,
+    pattern: /^(?:(?:http|https):\/\/)?(?:www\.)?(?:youtube\.com)\/(channel|user)\/([A-Za-z0-9-_\.]+)\/?$/im,
+  },
+  {
+    platform: SocialMediaPlatform.WEBSITE,
+    pattern: /^(?:(?:http|https):\/\/)?(?:www\.)?(?:[a-z0-9](?:[a-z0-9-]{0,61}[a-z0-9])?\.)+[a-z0-9][a-z0-9-]{0,61}[a-z0-9]\/?$/im,
+  },
+]
```

Comment:
Thanks a lot for being so thorough with these platform-specific checks! This is greatly appreciated